peakinvestigator/lib/uploader.py

```diff
@@ -0,0 +1,94 @@
+## -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016, Veritomyx, Inc.
+#
+# This file is part of the Python SDK for PeakInvestigator
+# (http://veritomyx.com) and is distributed under the terms
+# of the BSD 3-Clause license.
+from peakinvestigator.lib import *
+
+class Uploader(object):
+
+    def __init__(self, host, token, progress_factory, log_level=logging.WARNING):
+        self.logger = logging.getLogger('Uploader')
+        self.logger.setLevel(log_level)
+        self.progress_factory = progress_factory
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.ssl_sock = ssl.wrap_socket(self.sock)
+        server_address = (socket.gethostbyname(host), 9999)
+
+        self.ssl_sock.connect(server_address)
+        self.ssl_sock.sendall(token.encode('utf-8'))
+        response = self.ssl_sock.recv(4096)
+        self.logger.info('Server response:', response)
+
+        if response != 'Authorized'.encode('utf-8'):
+            raise Exception('Client not authorized')
+
+        self.buffer = bytearray(4096)
+
+        global Progress
```

Comment:
Global variables are generally super-bad because it makes tracking down bugs very difficult.

Reply:
Please just create a new commit without the use of global and push that. It'll automatically update the diffs here on Github, making it very easy to review.

Reply:
New commit is added. Progress_factory is now defined in user script.

Reply:
Did you push the commit? I don't see anything new at https://github.com/Veritomyx/PeakInvestigator-Python-SDK/pull/14/commits.

Reply:
I created a new branch and pull request to remove earlier issues.
https://github.com/Veritomyx/PeakInvestigator-Python-SDK/pull/15